src/data/repositories/IncidentActionD2Repository.ts

```diff
@@ -178,13 +179,34 @@ export class IncidentActionD2Repository implements IncidentActionRepository {
                     if (saveResponse.status === "ERROR" || !diseaseOutbreakId) {
                         return Future.error(new Error(`Error saving Incident Action`));
                     } else {
+                        if (formOptionsToDelete && formOptionsToDelete.length > 0) {
+                            return this.deleteIncidentResponseAction(formOptionsToDelete);
+                        }
+
                         return Future.success(undefined);
                     }
                 });
             });
         });
     }
 
+    private deleteIncidentResponseAction(events: Id[]): FutureData<void> {
+        return apiToFuture(
+            this.api.tracker.post(
+                { importStrategy: "DELETE" },
+                // an array of event ids is sufficient for TrackerPostRequest with DELETE importStrategy
+                // @ts-ignore
```

Comment:
(not need to change) In these cases another thing we usually do is to create the D2TrackerEvent like this:

```
        const d2Events: D2TrackerEvent[] = events.map(event => ({
            event: event,
            status: "COMPLETED",
            program: RTSL_ZEBRA_PROGRAM_ID,
            orgUnit: RTSL_ZEBRA_ORG_UNIT_ID,
            occurredAt: "",
            dataValues: [],
        }));
```

Reply:
oh i see. noted!